File: 10_find-the-index-of-the-first-occurrence-in-a-string/solution.py
def strStr(haystack, needle):
    """
    :type haystack: str
    :type needle: str
    :rtype: int
    """

    for i in range(len(haystack)):
        if haystack[i:len(needle)+i] == needle:
            return i
    
    return -1


res = strStr(haystack="sadbut" , needle ="sad")
print(res)

File: 10_find-the-index-of-the-first-occurrence-in-a-string/solution.cpp
#include <string>
#include <iostream>
using namespace std;

int strStr(string haystack, string needle) {
    for(int i = 0; i < haystack.length();i++) {
        if (haystack.substr(i,needle.length()) == needle) {
            return i;
        }
    }
    return -1;
}

int main() {
    string haystack = "example";
    string needle = "xa";
    int result = strStr(haystack, needle);
    std::cout << result << std::endl;
    return 0;
}